### ISSUE TITLE
pass4SymmKey setting causes restarts of splunk

### ISSUE THREAD

**Issue by rbclark** [voxpupuli/puppet-splunk#197]
## Affected Puppet, Ruby, OS and module versions/distributions

    Puppet: 4.10.6
    Ruby: 2.0.0p648
    Distribution: RedHat
    Module version: 7.2.0

## How to reproduce (e.g Puppet code you use)

```puppet
splunk_server { 'clustering/pass4SymmKey':
  value => "password",
}
```

## What are you seeing

The password is set and splunk automatically encrypts it using its `splunk.secret`, however splunk and puppet then begin fighting over the password line, with puppet changing it back to the unencrypted password and restarting splunk (and then splunk changing it back to the encrypted password).

## What behaviour did you expect instead

The password to be set and not keep changing back and forth, and splunk to not restart every run.

## Any additional information you'd like to impart

Unfortunately, it seems that splunk attempts to hold their method of encrypting files very close, so implementing the encryption of the file inside of puppet may be difficult. I have had a few thoughts about generating a shasum of the password and using that to verify the password hasn't changed since last run, but even that is difficult since technically the server.secret can change and the password would then not be updated accordingly (unless puppet is also managing the server.secret).

**Comment by rbclark** [comment]
I'm not sure if there is a reason this isn't the case already, however this may be the best argument I can make for implementing the `puppet_inifile` refreshonly parameter. This would mean that I could write some hashed value of my password to a file, and subscribe to it with the `clustering/pass4SymmKey` resource, which seems like a bit of a hack but would probably work.

**Comment by rbclark** [comment]
I'm not sure if there is a reason this isn't the case already, however this may be the best argument I can make for implementing the `puppet_inifile` refreshonly parameter. This would mean that I could write some hashed value of my password to a file, and subscribe to it with the `clustering/pass4SymmKey` resource, which seems like a bit of a hack but would probably work.

**Comment by kenyon** [comment]
Closing as fixed by #362.
